pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1019,6 +1051,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(node *model.
 						}
 					} else {
 						// Standard logic for headless and non headless services
+						if servicePort.Protocol.IsThrift() {
+							listenerOpts.bind = service.Address
```

Comment:
?? why do you need this?

Reply:
thrift like grpc (and like http) essentially will have the destination as part of the rpc call. i.e. envoy can listen on 0.0.0.0:port and demux among various destinations if needed right? If so, remove this check. Instead, on top of this function where we determine the listener's bind address by checking if the protocol is http or tcp, add a check for thrift as well. if thrift or http, you can have the bind on 0.0.0.0. there is no need to have a listener per service port as it simply adds to the number of listeners you shipto envoy.

Reply:
Well, it depends. The Thrift RouteMatch filter [requires _one of `method_name`, `service_name` to be set_](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/network/thrift_proxy/v2alpha1/route.proto#config-filter-network-thrift-proxy-v2alpha1-routematch). In our case, method name is set, but it's not prefixed with the service name (or namespaced in any way), so I don't think it would be possible for the thrift filter to match a request to a specific backend.

Reply:
In general, if thrift filter is disabled, the configuration should not be different from what we have today. This code piece doesn’t follow that principle. I suggest you look at all the if (thrift) do x points in your PR to ensure that this principle is followed. We need this because thrift is not like MySQL. It has its own stack.